src/api/objects/LegacyObjectAPIInterceptor.js

```diff
@@ -1,15 +1,56 @@
 define([
     './object-utils',
-    './ObjectAPI'
+    './ObjectAPI',
+    './objectEventEmitter'
 ], function (
     utils,
-    ObjectAPI
+    ObjectAPI,
+    objectEventEmitter
 ) {
-    function ObjectServiceProvider(objectService, instantiate) {
+    function ObjectServiceProvider(objectService, instantiate, topic) {
         this.objectService = objectService;
         this.instantiate = instantiate;
+
+        this.topicService = topic;
+        this.generalTopic = topic('mutation');
+        this.bridgeEventBuses();
     }
 
+    /**
+     * Bridges old and new style mutation events to provide compatibility between the two APIs
+     * @private
+     */
+    ObjectServiceProvider.prototype.bridgeEventBuses = function () {
+        var generalTopicListener;
+
+        var handleMutation = function (newStyleObject) {
+            var keyString = utils.makeKeyString(newStyleObject.key);
+            var oldStyleObject = this.instantiate(utils.toOldFormat(newStyleObject), keyString);
+            var specificTopic = this.topicService("mutation:" + keyString);
+
+            // Don't trigger self
+            if (generalTopicListener){
+                generalTopicListener();
+            }
+            this.generalTopic.notify(oldStyleObject);
+            specificTopic.notify(oldStyleObject.getModel());
```

Comment:
@larkin good point, I wonder if it will also be necessary to fetch from the model cache here, otherwise we're working against a different instance of DomainObject __and__ model. Fetching from model cache is synchronous, but not necessarily guaranteed to return anything?